### PR TITLE
Pubcommon id and shared id submodules: merge into one submodule

### PR DESCRIPTION
**Pull request by NothingIsMatter**
## Type of change
- [ x] Feature

## Description of change
Merge pubcid and sharedid
## Other information
Issue - https://github.com/prebid/Prebid.js/issues/6640

### CODE REVIEW

**Comment by patmmccann** [comment]
@NothingIsMatter prebid 5 was updated just now, can you repull master / merge conflicts?

**Comment by jdwieland8282** [comment]
@patmmccann, @NothingIsMatter is in the EU, is this something that needs to be addressed today, or can it wait until Monday?

**Comment by patmmccann** [comment]
It can definitely wait,  the 5.0 release is two weeks out

File: modules/sharedIdSystem.js [hunk 1]
```diff
@@ -1,373 +1,107 @@
...
+   *
+   * On the other hand, if there is no pixelUrl, then the extendId should return storedId so that
+   * its expiration time is updated.  Sharedid, however, will have to be updated by this submodule
+   * separately.
+   *
+   * @function
+   * @param {SubmoduleParams} [config]
+   * @param {ConsentData|undefined} consentData
+   * @param {Object} storedId existing id
+   * @returns {IdResponse|undefined}
```
Comment (mmoschovas) [inline modules/sharedIdSystem.js:103]: Looks like the extendId function comments are included but the pubCommonId function itself was not placed in here. Was this intentional? The domainOverride function was also not included as it was in the last PR

File: modules/sharedIdSystem.js [hunk 2]
```diff
@@ -1,373 +1,107 @@
...
+   * @param {string} storedId Existing pubcommon id
+   * @returns {IdResponse}
    */
-  getId(config, consentData) {
-    const coppa = coppaDataHandler.getCoppa();
-    if (coppa) {
-      utils.logInfo('SharedId: IDs not provided for coppa requests, exiting SharedId');
-      return;
+  getId: function (config = {}, consentData, storedId) {
+    const {params: {create = true, pixelUrl} = {}} = config;
```
Comment (mmoschovas) [inline modules/sharedIdSystem.js:67]: Is the coppa condition no longer needed? It seems to have been dropped

File: test/spec/modules/sharedIdSystem_spec.js [hunk 3]
```diff
@@ -1,77 +0,0 @@
-import {
```
Comment (mmoschovas) [inline test/spec/modules/sharedIdSystem_spec.js:1]: Why is this module test being removed?

**Review by mmoschovas — CHANGES_REQUESTED** [review]
Few things to take a look at here. Also it seems as though perhaps some of the files were overlooked in this update. The previous change consisted of 31 files whereas this update has only modified 25. Just doing a quick comparison, I can see that openx and ix adapters no longer have the update to remove the sharedId references. I would suggest going over this again to make sure all necessary updates are included in this version as well

File: test/spec/modules/sharedIdSystem_spec.js (hunk 3 above)
Comment (NothingIsMatter) [reply test/spec/modules/sharedIdSystem_spec.js:1]: According to https://github.com/prebid/Prebid.js/issues/6640: 

> remove the following sharedid files
> 
> modules/sharedIdSystem.js
> test/spec/modules/sharedIdSystem_spec.js
> modules/sharedIdSystem.md

File: modules/sharedIdSystem.js (hunk 2 above)
Comment (NothingIsMatter) [reply modules/sharedIdSystem.js:67]: As I understand coppa is related to sharedId = and I need to remove all refernces to sharedId in this file, so I decided to remove coppa too

File: modules/sharedIdSystem.js (hunk 1 above)
Comment (NothingIsMatter) [reply modules/sharedIdSystem.js:103]: No, that wasn`t intentional, will remove it

**Comment by NothingIsMatter** [comment]
@mmoschovas added comments, updated the branch

File: test/spec/modules/sharedIdSystem_spec.js (hunk 3 above)
Comment (NothingIsMatter) [reply test/spec/modules/sharedIdSystem_spec.js:1]: updated

File: modules/sharedIdSystem.js (hunk 1 above)
Comment (NothingIsMatter) [reply modules/sharedIdSystem.js]: updated

File: modules/sharedIdSystem.js (hunk 2 above)
Comment (NothingIsMatter) [reply modules/sharedIdSystem.js:73]: updated

**Comment by mmoschovas** [comment]
@NothingIsMatter I think there has been some confusion here most likely due to the nature of the ticket but quite possibly because of my comments. I just want to make sure there is a shared understanding here:

1) There are two pubCommon files - pubCommonId and pubCommonIdSystem. My understanding as per Prebid docs is that pubCommonIdSystem is currently used and pubCommonId is deprecated. Meaning that the purpose of this PR is to basically transfer code to sharedIdSystem and its corresponding unit tests from the pubCommonIdSystem. It seems the test here have been taken from the pubCommonId file. It looks as though perhaps there was never a designated test for the  pubCommonIdSystem, so these tests will need to be created under sharedIdSystem_spec since I dont see one existing currently

2) In terms of the comments and references, I am not seeing this having been fully updated. For clarification, I am referring to areas of the code such as the following. Effectively the pubCommonId module has been renamed to sharedId and this should be reflected in these areas as to be consistent; otherwise, this can become very confusing.

```
* This module adds PubCommonId to the User ID module 
* The {@link module:modules/userId} module is required 
* @module modules/pubCommonIdSystem 
* @requires module:modules/userId
```

```
...
export const pubCommonIdSubmodule = {
...
```

```
...
submodule('userId', pubCommonIdSubmodule);
```

3) I just wanted to make sure you saw my previous note regarding some adapters that were not included in this PR that were in the previous PR. i.e. ix and openx (looks like there are a few more though). Just want to confirm that these adapters are being updated if their 5.0 branch code still includes the sharedId reference

**Comment by NothingIsMatter** [comment]
> @NothingIsMatter I think there has been some confusion here most likely due to the nature of the ticket but quite possibly because of my comments. I just want to make sure there is a shared understanding here:
> 
> 1. There are two pubCommon files - pubCommonId and pubCommonIdSystem. My understanding as per Prebid docs is that pubCommonIdSystem is currently used and pubCommonId is deprecated. Meaning that the purpose of this PR is to basically transfer code to sharedIdSystem and its corresponding unit tests from the pubCommonIdSystem. It seems the test here have been taken from the pubCommonId file. It looks as though perhaps there was never a designated test for the  pubCommonIdSystem, so these tests will need to be created under sharedIdSystem_spec since I dont see one existing currently
> 2. In terms of the comments and references, I am not seeing this having been fully updated. For clarification, I am referring to areas of the code such as the following. Effectively the pubCommonId module has been renamed to sharedId and this should be reflected in these areas as to be consistent; otherwise, this can become very confusing.
> 
> ```
> * This module adds PubCommonId to the User ID module 
> * The {@link module:modules/userId} module is required 
> * @module modules/pubCommonIdSystem 
> * @requires module:modules/userId
> ```
> 
> ```
> ...
> export const pubCommonIdSubmodule = {
> ...
> ```
> 
> ```
> ...
> submodule('userId', pubCommonIdSubmodule);
> ```
> 
> 1. I just wanted to make sure you saw my previous note regarding some adapters that were not included in this PR that were in the previous PR. i.e. ix and openx (looks like there are a few more though). Just want to confirm that these adapters are being updated if their 5.0 branch code still includes the sharedId reference

1.  I did PR according to https://github.com/prebid/Prebid.js/issues/6640, it doesn`t contains any info regarding pubCommonId.js, hence i did not change it at all, also according to issue, I renamed PubcommonIdSystem_spec to sharedIdSystem_spec without changing it, but as you said its relates to pubCommonId, not to pubCommonIdSystem, probably need some input here from @jdwieland8282 
2. Ok, I will change it
3. Yes, probably I missed some of the changes, will fix it too.

**Comment by jdwieland8282** [comment]
@mmoschovas mind re reviewing we've made the requested changes.

**Comment by mmoschovas** [comment]
@jdwieland8282 are the unit tests going to be updated as well? It still looks like the spec file is testing the deprecated pubCommonId file

File: test/spec/modules/userId_spec.js [hunk 4]
```diff
@@ -25,7 +25,6 @@ import {
 import {server} from 'test/mocks/xhr.js';
 import find from 'core-js-pure/features/array/find.js';
 import {unifiedIdSubmodule} from 'modules/unifiedIdSystem.js';
-import {pubCommonIdSubmodule} from 'modules/pubCommonIdSystem.js';
```
Comment (SKOCHERI) [inline test/spec/modules/userId_spec.js:28]: @mmoschovas pubCommonIdSystem.js is removed from the tests.

Comment (mmoschovas) [reply test/spec/modules/userId_spec.js:28]: @SKOCHERI I'm referring to test/spec/modules/sharedIdSystem_spec.js  which is using the pubCommonId.js functions not sharedIdSystem.js

Comment (SKOCHERI) [reply test/spec/modules/userId_spec.js:28]: @mmoschovas In this PR, the pubCommonIdSystem.js is deleted but pubCommonId.js  is as is.

Comment (SKOCHERI) [reply test/spec/modules/userId_spec.js:28]: @mmoschovas The test/spec/modules/sharedIdSystem_spec.js is updated. Can you review it?

**Review by mmoschovas — APPROVED** [review]
LGTM

**Review by jlukas79 — APPROVED** [review]

**Comment by patmmccann** [comment]
@NothingIsMatter merge conflict showed up

**Review by smenzer — APPROVED** [review]
LGTM

**Pull request merged**